client/my-sites/themes/index.web.js

```diff
@@ -25,14 +26,17 @@ export default function( router ) {
 				router( '/themes/upload', siteSelection, sites, makeLayout );
 				router( '/themes/upload/:site_id?', siteSelection, upload, navigation, makeLayout );
 			}
+
 			const loggedInRoutes = [
+				`/themes/${ langRouteParams }`,
```

Comment:
Do we want to override the user preference language when logged in?

Reply:
Thanks so much for looking at this, @seear ! 

I think something is completely flying over my head.

I added a route catch the lang here because, after much restarting and refreshing, I noticed that on a hard refresh on http://calypso.localhost:3000/themes, even if I'm logged in, the [node router entry](https://github.com/Automattic/wp-calypso/pull/24561/files#diff-750f44a78dda5cf536254ebb29b69f5eR29) will pick it up first. At this point I can't find a way to check for a logged-in state initially, and there's nothing in `context.store.getState()` to help me,

This leads me to the next conundrum...

> it might be worth thinking about saving the lang as state to fix the refresh problem

The lang is being saved to state, but it isn't available at render time after I refresh the page. 

I did try dispatching a [setLocale action](https://github.com/Automattic/wp-calypso/blob/e930e16/client/state/ui/language/actions.js
) via `context.store.dispatch()`, but it doesn't persist between hard refreshes. 

The NUX sign up flow seems to get around this by storing the [locale into localeStorage](https://github.com/Automattic/wp-calypso/blob/e930e16/client/signup/controller.js#L80).

I'm pretty sure I'm missing something completely obvious, any tips you have will be _greatly_ appreciated :)

Reply:
> I think something is completely flying over my head.

I'll try and shed some light on some of this.

> I added a route catch the lang here because, after much restarting and refreshing, I noticed that on a hard refresh on http://calypso.localhost:3000/themes, even if I'm logged in, the [node router entry](https://github.com/Automattic/wp-calypso/pull/24561/files#diff-750f44a78dda5cf536254ebb29b69f5eR29) will pick it up first.

Yeah, that's expected -- hard refresh means force a fresh fetch from the server, so our node server will get a request here. 

> At this point I can't find a way to check for a logged-in state initially, and there's nothing in `context.store.getState()` to help me

The server side _can_ determine whether a user is logged-in or not by looking at the auth cookie (which we access mostly through the legacy `lib/user` or its Redux counterpart, e.g. the `getCurrentUserId()` selector). However, since that cookie comes from WordPress.com, this technique unfortunately doesn't work in our local development environments. This is an annoying shortcoming that leads e.g. to the rendering glitch in calypso.localhost:3000/themes when logged into WP.com.

Reply:
> I did try dispatching a [setLocale action](https://github.com/Automattic/wp-calypso/blob/e930e16/client/state/ui/language/actions.js
) via `context.store.dispatch()`, but it doesn't persist between hard refreshes. 

As of ..., we should be persisting state when logged-out. I noticed that the `localeSlug` reducer uses a [schema](https://github.com/Automattic/wp-calypso/blob/e930e16e0667ab4ffcf85ada02912a08db0245be/client/state/ui/language/reducer.js#L25), so it should be persisted. I think it's possible that we're seeing a bug here that causes re-hydrated server state to overwrite persisted client state which is also causing trouble with #24335. I'll comment on a possible fix there. 

> The NUX sign up flow seems to get around this by storing the [locale into localeStorage](https://github.com/Automattic/wp-calypso/blob/e930e16/client/signup/controller.js#L80).

I'd rather avoid a workaround like that.